File: solved-ac/class-2/15829/lua/15829.lua
local r = 31
local m = 1234567891

local length = io.read("*n")
io.read("*l") -- \n 자르기
local str = io.read("*a")

local sum = 0
local power = 1
for i = 1, length, 1 do
    sum = (sum + ((string.byte(str, i) - 97 + 1) * power)) % m
    power = (power * r) % m
end

print(sum)

File: solved-ac/class-2/15829/c/15829.c
#include <stdio.h>
#include <stdlib.h>

#define r 31ll
#define m 1234567891ll

int main() {
    int length;
    scanf("%d", &length);

    long long sum = 0;
    long long power = 1;
    for(int i = 0; i < length; i++) {
        char c;
        scanf(" %c", &c);

        sum = (sum + ((c - 'a' + 1) * power)) % m;
        power = (power * r) % m;
    }

    printf("%lld", sum);

    return 0;
}